src/query_backend.h

```diff
@@ -377,6 +377,13 @@ class QueryableBackend {
 class FullBackend: public QueryableBackend, public RecBackend {
  public:
   virtual ~FullBackend() {}
+
+  /// Closes the backend, if approriate.
+  virtual void Close() {}
```

Comment:
This would probably better to in the RecBackend class since it is mostly important for (all) cases where we are writing to the db.